baseapp/baseapp.go

```diff
@@ -481,6 +483,9 @@ func (app *BaseApp) getContextForTx(mode runTxMode, txBytes []byte) (ctx sdk.Con
 		WithVoteInfos(app.voteInfos).
 		WithConsensusParams(app.consensusParams)
 
+	if mode == runTxModeReCheck {
+		ctx = ctx.WithIsReCheckTx(true)
```

Comment:
we should return here
```suggestion
		return ctx.WithIsReCheckTx(true)
```

Reply:
nit: I'd also remove the named `ctx` on the result of the function (L480)

Reply:
No need to?